test/dtypes.c

```diff
@@ -7567,6 +7567,177 @@ test_named_indirect_reopen(hid_t fapl)
     return 1;
 } /* end test_named_indirect_reopen() */
 
+/*-------------------------------------------------------------------------
+ * Function:    test_named_indirect_reopen_file
+ *
+ * Purpose: Tests that a named compound datatype that refers to a named
+ *          string datatype can be reopened indirectly through H5Dget_type,
+ *          and shows the correct H5Tcommitted() state, including after the
+ *          file has been closed and reopened.
+ *
+ * Return:  Success:    0
+ *          Failure:    number of errors
+ *
+ * Programmer:  Mark Hodson
+ *              Tuesday, March 17, 2015
+ *
+ *-------------------------------------------------------------------------
+ */
+static int
+test_named_indirect_reopen_file(hid_t fapl)
+{
+    hid_t          file             = H5I_INVALID_HID;
+    hid_t          space            = H5I_INVALID_HID;
+    hid_t          cmptype          = H5I_INVALID_HID;
+    hid_t          reopened_cmptype = H5I_INVALID_HID;
+    hid_t          strtype          = H5I_INVALID_HID;
+    hid_t          reopened_strtype = H5I_INVALID_HID;
+    hid_t          dset             = H5I_INVALID_HID;
+    static hsize_t dims[1]          = {3};
+    size_t         strtype_size, cmptype_size;
+    char           filename[1024];
+
+    TESTING("indirectly reopening recursively committed datatypes including file reopening");
+
+    /* PREPARATION */
+
+    /* Create file, dataspace */
+    h5_fixname(FILENAME[1], fapl, filename, sizeof filename);
+    if ((file = H5Fcreate(filename, H5F_ACC_TRUNC, H5P_DEFAULT, fapl)) < 0)
+        TEST_ERROR;
+    if ((space = H5Screate_simple(1, dims, dims)) < 0)
+        TEST_ERROR;
+
+    /* Create string type */
+    if ((strtype = H5Tcopy(H5T_C_S1)) < 0)
+        TEST_ERROR;
+    if (H5Tset_size(strtype, H5T_VARIABLE) < 0)
+        TEST_ERROR;
+
+    /* Get size of string type */
+    if ((strtype_size = H5Tget_size(strtype)) == 0)
+        TEST_ERROR;
+
+    /* Commit compound type and verify the size doesn't change */
```

Comment:
This commits the string type, not the compound type

Reply:
Corrected.